python/ray/tests/test_task_events.py

```diff
@@ -118,3 +120,140 @@ def verify():
         timeout=10,
         retry_interval_ms=500,
     )
+
+
+def test_fault_tolerance_parents_finish(shutdown_only):
```

Comment:
can you add 2 more cases?

1. Test with actors (nested actors) and actor tasks.
2. Test a complicated e2e case. One idea;
- Start a big tree that has both tasks/actors/actor tasks (that has a height that's higher than 4). 
- Randomly fail one or two of them and make sure all children are marked as failure properly
- Loop these processes 2~3 times.